edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/handlers/PolicyUpdateHandler.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    using System;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Hub.Core;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Config;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// Responsible for listening for EdgeHub config updates (twin updates)
+    /// and pushing new authorization policy definition from EdgeHub config to the Mqtt Broker.
+    /// </summary>
+    public class PolicyUpdateHandler : IMessageProducer
+    {
+        // !Important: please keep in sync with mqtt-edgehub::command::POLICY_UPDATE_TOPIC
+        const string Topic = "$internal/authorization/policy";
+
+        readonly Task<IConfigSource> configSource;
+
+        IMqttBrokerConnector connector;
+
+        public PolicyUpdateHandler(Task<IConfigSource> configSource)
+        {
+            this.configSource = configSource;
+        }
+
+        public void SetConnector(IMqttBrokerConnector connector)
+        {
+            this.connector = connector;
+            this.connector.OnConnected += async (sender, args) => await this.OnConnect();
+        }
+
+        async Task OnConnect()
+        {
+            var configSource = await this.configSource;
+            configSource.ConfigUpdated +=
+                async (sender, config) => await this.ConfigUpdateHandler(config);
+
+            var config = await configSource.GetConfig();
+            await config.ForEachAsync(async config =>
+            {
+                var update = ConfigToPolicyUpdate(config);
+                await this.PublishPolicyUpdate(update);
+            });
+        }
+
+        async Task ConfigUpdateHandler(EdgeHubConfig config)
+        {
+            PolicyUpdate update = ConfigToPolicyUpdate(config);
+            await this.PublishPolicyUpdate(update);
+        }
+
+        static PolicyUpdate ConfigToPolicyUpdate(EdgeHubConfig config)
+        {
+            // TODO: conversion here.
```

Comment:
Convert from AuthorizationConfiguration to the PolicyUpdate?

Reply:
Yes, this will be added in the next PR.